pyclibrary/c_parser.py

```diff
@@ -700,6 +700,14 @@ def print_all(self, filename=None):
             else:
                 pprint(self.file_defs[filename][k])
 
+    def __str__(self):
```

Comment:
I can see the appeal in having str gives you a fully formatted representation of the parser. However I dislike the amount of code duplication between \_\_str\_\_ and print_all. Could you reformat the code using an helper method (possibly private) formatting a string but handling filename and use it in both  \_\_str\_\_ and print_all ?